packages/shared/test/unit/args_tests.js

```diff
@@ -0,0 +1,28 @@
+/* eslint-env jest */
+import { args, getArgName } from '../../args';
+
+describe('args(args)', () => {
+    it('it should parse the arguments correctly', () => {
+        const input = ['hello', 'world', '-c', 'config', '--filename', 'filename', 'otherargs', 'remaining'];
+        const expectedOutput = {
+            named: { c: 'config', filename: 'filename' },
+            positional: ['hello', 'world', 'otherargs', 'remaining']
+        };
+        expect(args(input)).toEqual(expectedOutput);
+    });
+
+    it('it should throw an error if value is not passed for arg', () => {
+        const input = ['hello', 'world', '-c'];
+        expect(() => args(input)).toThrowError();
+    });
+});
+
+describe('getArgName(arg)', () => {
+    it('it should extract the argument name for single character arg', () => {
+        expect(getArgName('-c')).toBe('c');
+    });
+
+    it('it should extract the argument name for word arg', () => {
+        expect(getArgName('--config')).toBe('config');
+    });
```

Comment:
Add a test case for `getArgName('--some-long-name')` this looks to be failing in the latest commit.

Reply:
This is implemented.